### PR TITLE
Update Cypher third party library dependencies

### PR DESCRIPTION
**Pull request by henriknyman**

### CODE REVIEW

File: community/cypher/cypher/pom.xml [hunk 1]
```diff
@@ -47,7 +47,7 @@
     <scala.binary.version>2.11</scala.binary.version>
     <maven.javadoc.failOnError>false</maven.javadoc.failOnError>
     <doclint-groups>none</doclint-groups>
-    <jetty.version>9.2.4.v20141103</jetty.version>
+    <jetty.version>9.3.5.v20151012</jetty.version>
```
Comment (Mats-SX) [inline community/cypher/cypher/pom.xml]: Could this be in `<scope test>` somehow?

Comment (pontusmelke) [inline community/cypher/cypher/pom.xml]: @Mats-SX it is in scope test, further down at line 326 

```
 <dependency>
      <groupId>org.eclipse.jetty</groupId>
      <artifactId>jetty-server</artifactId>
      <version>${jetty.version}</version>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>org.eclipse.jetty</groupId>
      <artifactId>jetty-servlet</artifactId>
      <version>${jetty.version}</version>
      <scope>test</scope>
    </dependency>
```

File: advanced/neo4j-advanced/LICENSES.txt [hunk 2]
```diff
@@ -260,7 +260,6 @@ SUCH DAMAGE.
 BSD License
   ASM Core
   Scala Compiler
-  scala-parser-combinators
```
Comment (pontusmelke) [inline advanced/neo4j-advanced/LICENSES.txt:263]: yay!

File: community/cypher/cypher/pom.xml (hunk 1 above)
Comment (MishaDemianenko) [inline community/cypher/cypher/pom.xml]: we have another jetty.version property defined at https://github.com/neo4j/neo4j/blob/3.0/community/server/pom.xml#L75
Why do you guys need another one?
If you can share the same version can you please move that property into common parent pom and reuse it?

**Pull request merged**